src/lease/mod.rs

```diff
@@ -122,17 +60,24 @@ impl LeaseKeepAliveTunnel {
                 };
                 match resp {
                     Ok(Some(resp)) => {
-                        resp_sender.send(Ok(From::from(resp))).unwrap();
+                        resp_sender
+                            .send(Ok(From::from(resp)))
+                            .await
+                            .unwrap_or_else(|_| panic!());
                     }
                     Ok(None) => {
                         return;
                     }
                     Err(e) => {
-                        resp_sender.send(Err(From::from(e))).unwrap();
+                        resp_sender
+                            .send(Err(From::from(e)))
+                            .await
+                            .unwrap_or_else(|_| panic!());
```

Comment:
proper panic and error handling!

`unwrap_or_else(|e| panic!("failed to send response, the error is: {}", e));`